forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt

```diff
@@ -5,15 +5,29 @@ import org.json.JSONObject
 sealed class ForageApiResponse<out T> {
     data class Success<out T>(val data: T) : ForageApiResponse<T>()
 
-    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>()
+    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>() {
+        companion object {
+            fun fromSQSError(sqsError: SQSError): Failure {
+                val forageError = ForageError(
+                    sqsError.statusCode,
+                    sqsError.forageCode,
+                    sqsError.message,
+                    sqsError.details
+                )
+                return Failure(listOf(forageError))
+            }
+        }
+    }
 }
 
 data class ForageError(
     val httpStatusCode: Int,
     val code: String,
-    val message: String
+    val message: String,
+    val details: JSONObject? = null,
 ) {
     override fun toString(): String {
+        // TODO: should we include the stringified JSON details in the toString() ?
```

Comment:
@dleis612 may have additional context, but my guess is that as long as the client doesn't parse the stringified ForageError, it is safe to do this

Would we have to update Android e2e QA tests too?

Reply:
I'm commenting now that I've had to investigate QA tests failing a few times. Updating the `.toString()` does not break the QA tests directly. The android mobile QA tests assert against `error.message` (see [here](https://github.com/teamforage/forage-android-sdk/blob/b62429777b2273eccb7509ca48fe924e3bed28e4/sample-app/src/main/java/com/joinforage/android/example/ui/complete/flow/payment/capture/FlowCapturePaymentViewModel.kt#L101) and [here](https://github.com/search?q=repo%3Ateamforage/mobile-qa-tests%20error_clauses&type=code)). This PR aims to display the full error (including the new `details`) field by introducing a new `<TextView />` that displays the whole thing and leaving the original `snapResponse` and `cashResponse` `<TextView / >`s untouched. Lmk if that clarifies!